lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -401,6 +443,58 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
                         scf::ExecuteRegionOp executeRegionOp) const;
   LogicalResult buildOp(PatternRewriter &rewriter, CallOp callOp) const;
 
+  // Sets up the necessary state and resources for a `CmpIOp` in
+  // `buildLibraryBinaryPipeOp` if `cmpIOp` has sequential logic based on its
+  // operands.
+  void setupCmpIOp(PatternRewriter &rewriter, CmpIOp cmpIOp, Operation *group,
+                   calyx::RegisterOp &condReg, calyx::RegisterOp &resReg,
+                   bool &isSeqCondCheck) const {
+    isSeqCondCheck = (calyx::isSeqLibOpRes(cmpIOp.getLhs()) ||
```

Comment:
You call this at least twice. Does it make sense to store this in a local variable, e.g.,

```cpp
bool lhsIsSeqOp = calyx::isSeqLibOpRes(cmpIOp.getLhs());
bool rhsIsSeqOp = calyx::isSeqLibOpRes(cmpIOp.getRhs());

...

assert(lhsIsSeqOp != rhsIsSeqOp && "...");
```

Reply:
sure!